pydap_extras/handlers/pcic.py

```diff
@@ -219,101 +253,83 @@ def get_full_query(self, stn_id, sesh):
         raise NotImplementedError
 
     def get_vars(self, stn_id, sesh):
-        raise NotImplementedError
+        """
+        Retrieves all variables of the specified type for a particular station.
+        The class variable `var_type` specifies the variable type. It must be given
+        a value in subclasses.
+        """
+        if not hasattr(self.__class__, "var_type"):
+            raise ValueError("This class must specify the class value 'var_type'")
+
+        q = (
+            sesh.query(Variable)
+            .join(VarsPerHistory, VarsPerHistory.vars_id == Variable.id)
+            .join(History, History.id == VarsPerHistory.history_id)
+            .join(Station, Station.id == History.station_id)
+            .join(Network, Network.id == Station.network_id)  # Or join to Variable
+            .filter(Station.id == stn_id)
+            .filter(variable_tags.contains(array([self.__class__.var_type])))
+        )
+        return [
+            (
+                x.name,
+                x.unit,
+                x.standard_name,
+                x.cell_method,
+                x.description,
+                x.display_name,
+            )
+            for x in q.all()
+        ]
 
 
+# TODO: RawPcicSqlHandler and ClimoPcicSqlHandler are very very similar. DRY up?
 class RawPcicSqlHandler(PcicSqlHandler):
     """Subclass of PcicSqlHandler which handles the raw observations"""
 
     extensions = re.compile(r"^.*\.rsql$", re.IGNORECASE)
     suffix = ".rsql"
     virtual = True
+    var_type = "observation"
 
     def get_full_query(self, stn_id, sesh):
-        """Sends a special query to the database that actually retrieves generated SQL for constructing an observation table (time by variable) for a single station. The query needs to return at least one column (obs_time) with additional columns for each available variable, if any. Uses the ``query_one_station`` stored procedure.
+        """
+        Retrieves generated SQL for constructing an observation table (time by variable)
+        for a single station. The query needs to return at least one column (obs_time)
+        with additional columns for each available variable, if any. Uses the
+        ``query_one_station`` stored procedure.
+
         :param stn_id: the *database* station_id of the desired station
         :type stn_id: int or str
-        :param sesh: an sqlalchemy session
+        :param sesh: a sqlalchemy session
         """
 
+        # TODO: What is this for? A comment at least would help.
```

Comment:
@jameshiebert , do you know why the following bit of code is here?

Reply:
VC attributes this to Cairo's work, but I didn't happen to see it at the time.

Based on the context, I'd guess that it was a kludge to avoid having the function completely fail on stations that existed/had metadata in the database (i.e. had a meta_station.native_id which is represented by the `stn_id` in this function), but didn't actually have any variables or observations. That's a situation that the data model supports, but that this code (apparently) didn't assume was possible.